python/nano/src/bigdl/nano/pytorch/inference/optimizer.py

```diff
@@ -179,19 +180,19 @@ def optimize(self, model: nn.Module,
                     input_sample = tuple(next(iter(training_data))[:-1])
                     try:
                         if accelerator is None and use_ipex is False:
-                            acce_model = model
+                            acce_model = deepcopy(model)
```

Comment:
Here I use deepcopy to make `self.optimized_model_dict["original"]["model"]` always in eval mode no matter whether original model in training mode.
If remove deepcopy, when user pass a model in training mode,  `self.optimized_model_dict["original"]["model"]` will also be training mode. So It cannot be guaranteed that the models saved by optimize are all eval model. Is this behavior reasonable?

Reply:
Will this be an issue if you are optimizing, say, BERT-Large? Will making a deep copy be a problem?

It is probably reasonable that the "original" model is the same as the original one?

Reply:
Users may get accelerated model from `optimizer.get_best_model(...)`, but if all accelerated method is not valid (either failed or does not meet user's accuracy requirement), we will let users get the original model.

Users may familiar this usage:
```python
acc_model = optimizer.get_best_model()
acc_model(x)
```

So, that's why we want to make the returned model in eval mode.

Well, it's just a minor point and it may does cause some mem issue. maybe we will just note our user the original model is the same reference as the model before optimization (i.e. they have the same training/eval mode)

Reply:
> Will this be an issue if you are optimizing, say, BERT-Large? Will making a deep copy be a problem in this case?
> 
> It is probably reasonable that the "original" model is the same as the original one?

Yeah, I agree. I will remove this deepcopy, and we will maybe note our user the original model is the same reference as the model before optimization as junwei comment.

Reply:
> Users may get accelerated model from `optimizer.get_best_model(...)`, but if all accelerated method is not valid (either failed or does not meet user's accuracy requirement), we will let users get the original model.
> 
> Users may familiar this usage:
> 
> ```python
> acc_model = optimizer.get_best_model()
> acc_model(x)
> ```
> 
> So, that's why we want to make the returned model in eval mode.
> 
> Well, it's just a minor point and it may does cause some mem issue. maybe we will just note our user the original model is the same reference as the model before optimization (i.e. they have the same training/eval mode)

Maybe a simpler semantics is to let the user know that `InferenceOptimizer` will always set all models (including the original one) to `eval`?

Reply:
So we don't restore model's status at the end of `optimize`, and just remind user that all models will be set to `eval`？

Reply:
Yes, I think it's better than deepcopy; to the user, `optimize` will always set `eval`, which is easier to understand.

Reply:
Done.